src/shared/components/instances-definitions.ts

```diff
@@ -100,6 +97,36 @@ export interface RecommendedInstance {
   topics: Topic[];
 }
 
+// https://chrisengelsma.medium.com/3342e47b9448
+function uniqueFilter(value, index, self) {
+  return self.indexOf(value) === index;
+}
+
+export function availableLanguages() {
+  // domains of crawled instances
+  var domains = instance_stats.stats.instance_details.map(i => i.domain);
+  // language names from translations
+  var languageNames = new Map(LANGUAGES.map(l => [l.code, l.name]));
+
+  return (
+    RECOMMENDED_INSTANCES
+      // ignore instances that were not crawled
+      .filter(r => domains.includes(r.domain))
+      // take all unique languages defined in this file
+      .flatMap(r => r.languages)
+      .filter(uniqueFilter)
+      // ignore languages that we dont have the name of
+      .filter(l => languageNames.has(l))
+      // return language code and name
+      .map(l => {
+        return {
+          code: l,
+          name: languageNames.get(l),
+        };
```

Comment:
I think this might work instead of the filter(has) + last map:

```
.flatMap(l => {
  const entry = languageNames.get(l);
  return entry ? {code: entry[0], name: entry[1]};
}